packages/core/src/app/client/state-manager.ts

```diff
@@ -0,0 +1,105 @@
+import BigNum = require('bn.js')
+
+import { isValidTransaction, isValidVerifiedStateUpdate } from '../common/utils'
+
+import {
+  HistoryProof,
+  PluginManager,
+  PredicatePlugin,
+  Range,
+  StateDB,
+  StateManager,
+  StateQuery,
+  StateQueryResult,
+  StateUpdate,
+  Transaction,
+  VerifiedStateUpdate,
+} from '../../interfaces'
+
+/**
+ * StateManager that validates transactions and wraps and modifies StateDB as necessary.
+ *
+ * See: http://spec.plasma.group/en/latest/src/05-client-architecture/state-manager.html for more details.
+ */
+export class DefaultStateManager implements StateManager {
+  private stateDB: StateDB
+  private pluginManager: PluginManager
+
+  public constructor(stateDB: StateDB, pluginManager: PluginManager) {
+    this.stateDB = stateDB
+    this.pluginManager = pluginManager
+  }
+
+  public async executeTransaction(
+    transaction: Transaction
+  ): Promise<{ stateUpdate: StateUpdate; validRanges: Range[] }> {
+    const result = {
+      stateUpdate: undefined,
+      validRanges: [],
+    }
+
+    if (!isValidTransaction(transaction)) {
+      // log here?
+      return result
+    }
+
+    // Get Verified updates for range
+    const { start, end }: Range = transaction.stateUpdate.id
+    const verifiedUpdates: VerifiedStateUpdate[] = await this.stateDB.getVerifiedStateUpdates(
+      start,
+      end
+    )
+
+    for (const verifiedUpdate of verifiedUpdates) {
+      if (!isValidVerifiedStateUpdate(verifiedUpdate)) {
+        // log here?
+        continue
+      }
+
+      const {
+        start: verifiedStart,
+        end: verifiedEnd,
+      }: Range = verifiedUpdate.stateUpdate.id
+      if (
+        transaction.block !== verifiedUpdate.verifiedBlockNumber + 1 ||
```

Comment:
The check here should be removed as we no longer require transactions to include a block number.